llarp/net/route.cpp

```diff
@@ -399,35 +399,36 @@ namespace llarp::net
 #undef FREE
     return gateways;
 #elif __APPLE__
+    LogDebug("get gateways not on ", ifname);
     const auto maybe = GetIFAddr(ifname);
     if (not maybe.has_value())
       return gateways;
     const auto interface = maybe->toString();
     // mac os is so godawful man
-    FILE* p = popen("netstat -rn -f inet", "r");
+    FILE* p = popen("/usr/sbin/netstat -rn -f inet", "r");
     if (p == nullptr)
+    {
       return gateways;
+    }
     char* line = nullptr;
     size_t len = 0;
     ssize_t read = 0;
     while ((read = getline(&line, &len, p)) != -1)
     {
-      std::string line_str(line, len);
-      if (line_str.find("default") == 0)
+      const std::string line_str(line, len);
+      const auto parts = llarp::split(line_str, " "sv);
+      std::vector<std::string_view> parts_nonempty;
+      for (auto part : parts)
       {
-        line_str = line_str.substr(7);
-        while (line_str[0] == ' ')
-        {
-          line_str = line_str.substr(1);
-        }
-        const auto pos = line_str.find(" ");
-        if (pos != std::string::npos)
+        if (not part.empty())
```

Comment:
It seems that if you use `split_any(input_str, delimiter(s), bool trim)` with trim=true then you will get this behavior without having to check here for empty splits.

Reply:
meh

Reply:
split_any used